server/service/src/processors/transfer/invoice/common.rs

```diff
@@ -146,3 +153,48 @@ pub(crate) fn convert_invoice_line_to_single_pack(
         })
         .collect()
 }
+
+pub(crate) fn auto_verify_if_store_preference(
+    ctx: &ServiceContext,
+    inbound_shipment: &InvoiceRow,
+) -> Result<(), RepositoryError> {
+    if inbound_shipment.r#type != InvoiceType::InboundShipment {
+        return Ok(());
+    }
+
+    match inbound_shipment.status {
+        repository::InvoiceStatus::New
+        | repository::InvoiceStatus::Allocated
+        | repository::InvoiceStatus::Picked
+        | repository::InvoiceStatus::Verified
+        | repository::InvoiceStatus::Cancelled => return Ok(()),
+        repository::InvoiceStatus::Shipped
+        | repository::InvoiceStatus::Received
+        | repository::InvoiceStatus::Delivered => (), // proceed to check auto verify pref
+    };
+    let should_auto_verify = InboundShipmentAutoVerify {}
+        .load(&ctx.connection, Some(inbound_shipment.store_id.to_string()))
+        .map_err(|e| {
+            RepositoryError::as_db_error(
+                "Could not load inbound shipment auto verify preference",
+                e,
+            )
+        })?;
+
+    if should_auto_verify {
+        update_inbound_shipment(
+            ctx,
+            UpdateInboundShipment {
+                id: inbound_shipment.id.to_string(),
+                status: Some(UpdateInboundShipmentStatus::Verified),
+                ..Default::default()
+            },
+            Some(&inbound_shipment.store_id),
```

Comment:
All the context changing was so I could call this method with a ctx.

Unfortunately the ctx started at the beginning of the call chain can't know what store_id and user_id we'll need, so just tacked a store_id in here.

That said... I need to do user_ID too in some capacity, as right now it's not making an activity log correctly!

Reply:
I made another commit after my comment after realising the logging wasn't working correctly! I ended up just making the ctx `system_context` (so actually used now) rather than `basic_context` which includes the OMS user ID constant.